File: fulfillment-playwright-poc/pages/dsid_page.py
"""DSID column + Details panel locators (BSD-29870)."""

from __future__ import annotations

from playwright.sync_api import Locator, Page


class DsidPage:
    COLUMN_LABEL = "DSID"
    JOB_TYPE_VALUE = "MetadataOnlyDelivery"
    PARTNER_VALUE = "PP YT FSP UK"

    GRID_COLUMN_HEADER = (
        f"xpath=//th[@title='{COLUMN_LABEL}']"
        f" | //th[.//span[normalize-space()='{COLUMN_LABEL}']]"
    )
    GRID_ROW = "xpath=//table[@id='orderTable']//tbody//tr | //tbody//tr[contains(@class,'row')]"
    DETAILS_DSID_VALUES = (
        f"xpath=//*[contains(@class,'label') and normalize-space()='{COLUMN_LABEL}']"
        "/following-sibling::*[contains(@class,'value')]//span[contains(@class,'demandSystemId-value')]//span"
    )

    def __init__(self, page: Page) -> None:
        self.page = page

    def grid_row(self, one_based_index: int) -> Locator:
        return self.page.locator(self.GRID_ROW).nth(one_based_index - 1)

File: src/test/java/com/paramount/test/ff/pageobjects/DsidPage.java
package com.paramount.test.ff.pageobjects;

import com.synergy.core.driver.By;

/** Locators for BSD-29870 — DSID column validation (MetadataOnlyDelivery / PP YT FSP UK). */
public class DsidPage {

    public static final String COLUMN_LABEL = "DSID";
    public static final String JOB_TYPE_FILTER = "Job type";
    public static final String JOB_TYPE_VALUE = "MetadataOnlyDelivery";
    public static final String PARTNER_FILTER = "Partner";
    public static final String PARTNER_VALUE = "PP YT FSP UK";
    public static final String DETAILS_PANEL_LABEL = "DSID";

    private final TableView tableView = new TableView();
    private final HomePage homePage = new HomePage();

    public By tableViewButton() {
        return tableView.getTableViewButton();
    }

    public By detailsButton() {
        return homePage.DetailsPanelButton();
    }

    public By gridColumnHeader() {
        return By.XPath("//th[@title='" + COLUMN_LABEL + "']"
                + " | //th[.//span[normalize-space()='" + COLUMN_LABEL + "']]"
                + " | //th[contains(normalize-space(),'" + COLUMN_LABEL + "')]");
    }

    public By gridDataRows() {
        return By.XPath("//table[@id='orderTable']//tbody//tr"
                + " | //tbody//tr[contains(@class,'row')]");
    }

    public By gridRowByIndex(int oneBasedIndex) {
        return By.XPath("(//table[@id='orderTable']//tbody//tr | //tbody//tr[contains(@class,'row')])["
                + oneBasedIndex + "]");
    }

    public By detailsPanelDsidSection() {
        return By.XPath("//*[contains(@class,'details') or contains(@class,'panel')]"
                + "//*[contains(@class,'label') and normalize-space()='" + DETAILS_PANEL_LABEL + "']");
    }

    public By detailsPanelDsidValues() {
        return By.XPath("//*[contains(@class,'label') and normalize-space()='" + DETAILS_PANEL_LABEL + "']"
                + "/following-sibling::*[contains(@class,'value')]"
                + "//span[contains(@class,'demandSystemId-value')]//span"
                + " | //*[contains(@class,'label') and normalize-space()='" + DETAILS_PANEL_LABEL + "']"
                + "/ancestor::div[contains(@class,'field') or contains(@class,'row')][1]"
                + "//*[contains(@class,'value')]//span[contains(@class,'demandSystemId-value')]"
                + " | //*[contains(@class,'label') and normalize-space()='" + DETAILS_PANEL_LABEL + "']"
                + "/following-sibling::*[contains(@class,'value')]//span");
    }
}
